### DISCUSSION TITLE
Trying to change shortcut to open Scriptkit not working as expected

### DISCUSSION THREAD

**Discussion by AmodeusR** [johnlindquist/kit#1132] [Errors]
Shortcut change is having a weird unexpected behavior. I tried changing it to some different shortcuts but it didn't work the way I expected.  I tried changing the shortcut from `ctrl + ;` to `ctrl + shift + :` but it didn't work, then I tried to change to `ctrl + shift + ?` and also didn't work, but when I checked again the last command shortcut worked for some reason.

It's either keeping a register of the old shortcut, which I don't think that's really the problem, or it has something to do with the way it tracks the shortcut, not working properly with other keyboard types other than the english one.

**Comment by johnlindquist** [comment]
@AmodeusR What keyboard layout do you use? Script Kit uses a lib to attempt to account for various types of keyboard layouts, but it's been an impossible task to account for them all

**Reply by johnlindquist** [reply]
You can also see in the ~/.kit/logs/kit.log how the conversion is working.

Also, ~/.kit/db/shortcuts.json is where the main shortcut is store. You can change it there as well

**Reply by AmodeusR** [reply]
Portuguese ABNT2.

I just checked to make sure, and what happens is when I configure the shortcut to be `ctrl + shift + ?` I need to use `ctrl + shit + :` and then it works.

It may be confusing to understand the buttons I'm referring to, but I'm referring to the ones that's shown in the Scriptkit when it asks for a confirmation...

**Reply by AmodeusR** [reply]
I just checked and the shortcut that works is equivalent to the english keyboard. the configured shortcut is `ctrl shift ?` which is equivalent to `ctrl shift :` in my keyboard.
 If I were to change to the english keyboard, it would output "?" instead of outputing ":" in the portuguese keyboard. Hence, in fact it's basing the shortcut off of the english keyboard.

**Reply by johnlindquist** [reply]
Alright, thanks for letting me know. It's using this lib https://www.npmjs.com/package/native-keymap , I'm not sure why Portuguese is failing when Chinese, German, Dvorak and others are working. We had another report of a Portuguese layout not working here:

https://github.com/johnlindquist/kit/issues/1060

Are you on Windows too?

**Reply by AmodeusR** [reply]
Yes, I'm on Windows.
